### PR TITLE
VZ-9436 - add private registry support to the VZ CLI

### PR DESCRIPTION
**Pull request by desagar**
Adds --image-registry and --image-prefix flags to support private registry install and upgrade

### CODE REVIEW

File: tools/vz/cmd/helpers/testdata/verrazzano-platform-operator.yaml
Comment (fredtibbitts) [inline tools/vz/cmd/helpers/testdata/verrazzano-platform-operator.yaml:1]: We now have two operator.yaml files for testing, this one and `operator-file-fake.yaml` that are nearly identical, we should remove one of them. This is probably also something we can do in a follow-on PR.

File: tools/vz/cmd/helpers/vpo.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ import (
 	"github.com/verrazzano/verrazzano/pkg/k8sutil"
 	"github.com/verrazzano/verrazzano/pkg/semver"
 	"github.com/verrazzano/verrazzano/platform-operator/apis/verrazzano/v1beta1"
+	constants2 "github.com/verrazzano/verrazzano/platform-operator/constants"
```
[the remark this answers is not in the document]
Comment (desagar) [reply tools/vz/cmd/helpers/vpo.go]: done

File: tools/vz/cmd/helpers/operator_yaml.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+	for idx := range containers {
+		// Use indexing on the slice to get a reference to the container so we can edit it. By default
+		// range returns copies so our edits won't stick
+		container := containers[idx].(map[string]interface{})
+		ctrUpdated := updatePrivateRegistryOnContainer(container, imageRegistry, imagePrefix)
+		updated = updated || ctrUpdated
+		if container["name"] == "verrazzano-platform-operator" {
+			envUpdated := addRegistryEnvVarsToContainer(container, imageRegistry, imagePrefix)
+			updated = envUpdated
```
[the remark this answers is not in the document]
Comment (desagar) [reply tools/vz/cmd/helpers/operator_yaml.go]: yes, good catch!

File: tools/vz/cmd/helpers/operator_yaml.go [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+				vpoWebhookDeployIdx = idx
+			}
+		}
+	}
+	return vpoDeployIdx, vpoWebhookDeployIdx
+}
+
+// updatePrivateRegistryVPODeploy updates the private registry information in the
+// given verrazzano-platform-operator (or webhook) deployment YAML. Returns true if vpoDeploy was modified
+func updatePrivateRegistryVPODeploy(vpoDeploy *unstructured.Unstructured, imageRegistry string, imagePrefix string, addRegistryEnvVars bool) (bool, error) {
```
[the remark this answers is not in the document]
Comment (desagar) [reply tools/vz/cmd/helpers/operator_yaml.go:83]: yeah that would be nice. I'll look into it after this is merged.

File: tools/vz/cmd/helpers/operator_yaml.go [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+	}
+	return editedOperatorFile, nil
+}
+
+func findVPODeploymentIndices(objectsInYAML []unstructured.Unstructured) (int, int) {
+	vpoDeployIdx := -1
+	vpoWebhookDeployIdx := -1
+	for idx, yamlObj := range objectsInYAML {
+		if yamlObj.GetObjectKind().GroupVersionKind().Kind == "Deployment" {
+			if yamlObj.GetName() == "verrazzano-platform-operator" {
```
[the remark this answers is not in the document]
Comment (desagar) [reply tools/vz/cmd/helpers/operator_yaml.go]: done

File: tools/vz/cmd/helpers/operator_yaml.go [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+
+	vpoDeployUpdated, err := updatePrivateRegistryVPODeploy(vpoDeploy, imageRegistry, imagePrefix, true)
+	if err != nil {
+		return "", err
+	}
+	vpoWebhookDeployUpdated, err := updatePrivateRegistryVPODeploy(vpoWebhookDeploy, imageRegistry, imagePrefix, false)
+	if err != nil {
+		return "", err
+	}
+	// vpoWebhookDeployUpdated, err := updatePrivateRegistryWebhookDeploy(vpoWebhookDeploy, imageRegistry, imagePrefix)
```
[the remark this answers is not in the document]
Comment (desagar) [reply tools/vz/cmd/helpers/operator_yaml.go]: done

File: tools/vz/cmd/helpers/command.go [hunk 6]
```diff
@@ -85,8 +86,29 @@ func GetVersion(cmd *cobra.Command, vzHelper helpers.VZHelper) (string, error) {
...
+	var response string
+	scanner := bufio.NewScanner(vzHelper.GetInputStream())
+	fmt.Fprintf(vzHelper.GetOutputStream(), "%s [Y/n]: ", questionText)
+	if scanner.Scan() {
+		response = scanner.Text()
+	}
+	if err := scanner.Err(); err != nil {
+		return false, err
+	}
+	if response == "y" || response == "Y" {
```
Comment (davisgomes) [inline tools/vz/cmd/helpers/command.go:104]: If response is empty, don't we want a "Yes"? It seems like it will default to "No"

**Review by fredtibbitts — DISMISSED** [review]

**Review by davisgomes — DISMISSED** [review]

**Review by davisgomes — APPROVED** [review]

**Pull request merged**